ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -188,15 +185,30 @@ public static UdfInvoker compile(final Method method, final ClassLoader loader)
     final AnnotatedParameterizedType annotatedReturnType
         = (AnnotatedParameterizedType) method.getAnnotatedReturnType();
     final ParameterizedType type = (ParameterizedType) annotatedReturnType.getType();
-    final Type valueType = type.getActualTypeArguments()[0];
+    final Type inputType = type.getActualTypeArguments()[0];
     final Type aggregateType = type.getActualTypeArguments()[1];
+    final Type outputType = type.getActualTypeArguments()[2];
     if (!isTypeSupported((Class<?>) getRawType(aggregateType), SUPPORTED_UDF_TYPES)) {
```

Comment:
I think we need to check if the `outputType` is supported - do we need to check if the aggregate type is supported? We might not need to because it's never output anywhere... but we can leave that for another time, for now let's check all three.

Reply:
that might be overly restrictive (to check all three) ? imagine I have been wanting to write some sketch-based UDAFs for a long time, which depend on serializing what amounts to <some arbitrary binary data> between invocations, then restricting this to only known SQL-types would prevent such a function and make me sad

Reply:
I totally agree (hence my comment) but I think we can leave that (and the corresponding testing suite) to a future PR.

Reply:
+1 - that'd be a pretty complex change to our serializers etc (or introduction of a binary type). I wouldn't glob that into this.